google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/ClientSideStatementImpl.java

```diff
@@ -0,0 +1,209 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.jdbc;
+
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.jdbc.StatementResult.ResultType;
+import com.google.common.base.Preconditions;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Implementation of the {@link ClientSideStatement} interface. The instances of this class are
+ * imported from the file 'ClientSideStatements.json' in the resources folder.
+ */
+class ClientSideStatementImpl implements ClientSideStatement {
+
+  /**
+   * Statements that set a value, such as SET AUTOCOMMIT ON|OFF, must specify a {@link
+   * ClientSideSetStatementImpl} that defines how the value is set.
+   */
+  static class ClientSideSetStatementImpl {
+    /** The property name that is to be set, e.g. AUTOCOMMIT. */
+    private String propertyName;
+    /** The separator between the property and the value (i.e. '=' or '\s+'). */
+    private String separator;
+    /** Regex specifying the range of allowed values for the property. */
+    private String allowedValues;
+    /** The class name of the {@link ClientSideStatementValueConverter} to use. */
+    private String converterName;
+
+    String getPropertyName() {
+      return propertyName;
+    }
+
+    String getSeparator() {
+      return separator;
+    }
+
+    String getAllowedValues() {
+      return allowedValues;
+    }
+
+    String getConverterName() {
+      return converterName;
+    }
+  }
+
+  static class CompileException extends Exception {
+    private static final long serialVersionUID = 1L;
+    private final ClientSideStatementImpl statement;
+
+    CompileException(Throwable cause, ClientSideStatementImpl statement) {
+      super(cause);
+      this.statement = statement;
+    }
+
+    @Override
+    public String getMessage() {
+      return "Could not compile statement " + this.statement.name;
+    }
+  }
+
+  static class ExecuteException extends RuntimeException {
+    private static final long serialVersionUID = 1L;
+    private final ClientSideStatementImpl statement;
+    private final String sql;
+
+    private ExecuteException(Throwable cause, ClientSideStatementImpl statement, String sql) {
+      super(cause);
+      this.statement = statement;
+      this.sql = sql;
+    }
+
+    @Override
+    public String getMessage() {
+      return "Could not execute statement " + this.statement.name + " (" + sql + ")";
+    }
+  }
+
+  /** The name of this statement. Used in error and info messages. */
+  private String name;
```

Comment:
minor - can address later - we should clean up the whitespace here between variable declarations and javadoc